webapp/src/components/content/textElement.tsx

```diff
@@ -11,26 +13,56 @@ import {MarkdownEditor} from '../markdownEditor'
 
 import {contentRegistry} from './contentRegistry'
 
+import './textElement.scss'
+
 type Props = {
     block: ContentBlock
     readonly: boolean
 }
 
-const TextElement = (props: Props): JSX.Element => {
-    const {block, readonly} = props
+const BlockTitleMaxBytes = 65535 // Maximum size of a TEXT column in MySQL
+const BlockTitleMaxRunes = BlockTitleMaxBytes / 4 // Assume a worst-case representation
+
+const TextElement = ({block, readonly}: Props): JSX.Element => {
     const intl = useIntl()
 
+    const [isError, setIsError] = useState<boolean>(false)
+    const [blockTitle, setBlockTitle] = useState(block.title)
```

Comment:
Why store `blockTitle` in a `useState`? Couldn't we just use `props.title` where that's currently used?

Reply:
Initially, the `onblur` event for the input field made an API call that replaced the new block.title with the old one. The problem with this approach was that if the API call returned an error, the block.title would revert to the old data, causing the newly typed data to be lost. To ensure the new data is retained even after an error, the `blockTitle` state was introduced.